### DISCUSSION TITLE
Is future known_covariates needed to do the predict ?

### DISCUSSION THREAD

**Discussion by langslike** [autogluon/autogluon#2778] [Q&A] [answered]
I use autogluon to predict financial data,  and according to the official tutorial, I set ignore_time_index=True. When I run  TimeSeriesPredictor.predict, there is the warning:  Because `ignore_time_index=True`, the predictor will ignore the time index of `known_covariates`. Please make sure that `known_covariates` contain only the future values of the known covariates (and the past values are not included).
So I checked the source code, in timeseries/predictor.py, there is an example:
Examples
        --------
        >>> print(data)
                            target  promotion  price
        item_id timestamp
        A       2020-01-05      20          0   19.9
                2020-01-06      40          1    9.9
                2020-01-07      32          0   15.0
        B       2020-03-01      13          0    5.0
                2020-03-02      44          1    2.9
                2020-03-03      72          1    2.9
        >>> predictor = TimeSeriesPredictor(prediction_length=2, known_covariates_names=["promotion", "price"]).fit(data)
        >>> print(future_known_covariates)
                            promotion  price
        item_id timestamp
        A       2020-01-08          1   12.9
                2020-01-09          1   12.9
        B       2020-03-04          0    5.0
                2020-03-05          0    7.0
        >>> predictor.predict(data, known_covariates=future_known_covariates)
                            target
        item_id timestamp
        A       2020-01-08      30
                2020-01-09      27
        B       2020-03-04      17
                2020-03-05       8
That is, to do the predict, **I must have future_known_covariates**, but what if I only have known_covariates in the past and now, can I still use autogluon to do the predict? what changes are required in data preparation? do I need to create a lag in train known_covariates data to account for this?

**Comment by shchur** [comment, marked as answer]
`known_covariates` correspond to covariates that are actually known in the future, so these need to be provided for the entire forecast horizon at prediction time. We will add support for covariates that are known only in the past (`past_covariates`) in the upcoming release.

**Reply by langslike** [reply]
Cool，that would be very helpful.
